tests/Imgix.Tests/SrcSetTest.cs

```diff
@@ -559,5 +559,141 @@ public void HeightAndAspectRatioIncludesDPRParam()
+            int[] expected = {
+                100, 116, 134, 156, 182, 210, 244, 282,
+                328, 380, 442, 512, 594, 688, 798, 926,
+                1074, 1246, 1446, 1678, 1946, 2258, 2618,
+                3038, 3524, 4088, 4742, 5500, 6380, 7400 };
+
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void TargetWidths328to4088()
+        {
+            List<int> actual = UrlBuilder.GenerateTargetWidths(start: 328, stop: 4088);
+            // `GenerateTargetWidths` depends on the start value; consequently,
+            // values between `start` and `stop` may be a couple pixels above
+            // (and perhaps below) our defaults (if tolerance is default).
+            int[] expected = {
+                328, 380, 442, 512, 594, 688,
+                /* 798 */ 800,
+                /* 926 */ 928,
+                /* 1074 */ 1076, 1248, 1446, 1678,
+                1948, 2258,
+                /* 2620 */ 2620,
+                /* 3038 */ 3040,
+                /* 3524 */ 3526, 4088};
+
+            Assert.AreEqual(expected, actual);
+
+        }
+
+        [Test]
+        public void TargetWidths100to8192MaxTolerance()
+        {
+            List<int> actual = UrlBuilder.GenerateTargetWidths(tol: 100000000);
+            int[] expected = { 100, 8192 };
+
+            Assert.AreEqual(expected, actual);
+
+        }
+
+        [Test]
+        public void TestCustomSrcSetPairs()
+        {
+            UrlBuilder ub = new UrlBuilder(
+                "test.imgix.net",
+                signKey: null,
+                includeLibraryParam: false,
+                useHttps: true);
+
+            String srcset = ub.BuildSrcSet(
+                "image.jpg",
+                new Dictionary<string, string>(),
+                start: 328, stop: 328);
+
+            Assert.AreEqual("https://test.imgix.net/image.jpg?w=328 328w", srcset);
+
+            String srcset640to720 = ub.BuildSrcSet(
+                "image.jpg",
+                new Dictionary<string, string>(),
+                start: 640, stop: 720);
+
+            String expect640to720 =
+                "https://test.imgix.net/image.jpg?w=640 640w," +
+                "\nhttps://test.imgix.net/image.jpg?w=720 720w";
+
+            Assert.AreEqual(expect640to720, srcset640to720);
+
+            String expect100to108 =
+                "https://test.imgix.net/image.jpg?w=100 100w," +
+                "\nhttps://test.imgix.net/image.jpg?w=102 102w," +
+                "\nhttps://test.imgix.net/image.jpg?w=104 104w," +
+                "\nhttps://test.imgix.net/image.jpg?w=106 106w," +
+                "\nhttps://test.imgix.net/image.jpg?w=108 108w";
+
+            String srcset100to108 = ub.BuildSrcSet(
+                "image.jpg",
+                new Dictionary<string, string>(),
+                start: 100,
+                stop: 108,
+                tol: 1);
```

Comment:
...or a situation where we want to specify `tol` but not `start` or `stop`?

Reply:
The discussion here really says "More Overloaded Functions." Let's discuss Monday, but I think we could really benefit from more OL functions––it's a little clunky as it, these suggestions ought to smooth this out, THANKS!!!